modules/internal/ChapelIteratorSupport.chpl

```diff
@@ -118,8 +127,32 @@ module ChapelIteratorSupport {
       return _getIteratorZipInternal(x, 1);
   }
 
+  inline proc _getIteratorZip(type t: _tuple) {
+    if (!(isTupleType(t))) then
+      compilerError("cannot zip over a type that is not a homogeneous tuple");
+    else if (!(isEnumType(t(1)))) then
+      compilerError("cannot zip over a tuple of types that is not of enums");
```

Comment:
This looks like you're adding a couple of extra error cases.  Would you add a test to lock those error cases in?  Or do we have futures that already cover them?